### PR TITLE
build: add prod docker file for official builds

### PR DESCRIPTION
**Pull request by Roasbeef**

### CODE REVIEW

**Review by guggero — APPROVED** [review]
tACK, LGTM :tada:

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+COPY --from=builder /go/bin/tapcli /bin/
+COPY --from=builder /go/bin/tapd /bin/
+
+# Store the SHA256 hash of the binaries that were just produced for later
+# verification.
+RUN sha256sum /bin/tapd /bin/tapcli > /shasums.txt \
+  && cat /shasums.txt
+
+# Expose tapd ports (gRPC, REST).
+EXPOSE 10029 8089
```
Comment (ffranr) [inline Dockerfile:54]: How did we select this port number? Are we sure it's unique enough?

**Review by ffranr — APPROVED** [review]
Looks good to me :+1:

File: Dockerfile (hunk 1 above)
Comment (Roasbeef) [reply Dockerfile:54]: The RPC port number? We just kinda picked one as we usually do.....

Can verify there isn't any major overlap before going to mainnet

Comment (dstadulis) [reply Dockerfile:54]: captured in https://github.com/lightninglabs/taproot-assets/issues/312

**Pull request merged**